lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/DataLoadingContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Common.Events;
+using Org.Apache.REEF.IO.PartitionedData;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    internal class DataLoadingContext<T> : IObserver<IContextStart>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(DataLoadingContext<T>));
+
+        [Inject]
+        internal DataLoadingContext(IInputPartition<T> partition)
+        {
+            Console.WriteLine("ENTERED CONTEXT");
+        }
+
+        public void OnNext(IContextStart value)
+        {
```

Comment:
Usually we don't want to put heavy data loading in the constructor, in case something is wrong, that will result in injection error - very misleading. 

We can make Initialize() method in FileSystemInputPartition public. Call it in 
OnNext(IContextStart value) 
of the context handler. But that only download the data if we want to download from remote to local. 
Not sure if we want to put GetPartitionHandle() logic into Initialize() or constructor of FileSystemInputPartition(). You can call it from OnNext() of the context handler any way. It will end up the same thing as OnNext(IContextStart value)  will be called at beginning when the Context is created. 

If we put the first iterate in task, I don't know how useful this decouple is because in that case, data is downloaded to memory in task.